Modules/itertoolsmodule.c

```diff
@@ -2447,7 +2447,7 @@ product_reduce(productobject *lz, PyObject *Py_UNUSED(ignored))
     if (lz->stopped) {
         return Py_BuildValue("O(())", Py_TYPE(lz));
     } else if (lz->result == NULL) {
-        return Py_BuildValue("OO", Py_TYPE(lz), lz->pools);
+        return PyTuple_Pack(2, Py_TYPE(lz), lz->pools);
```

Comment:
Please remove this edit.  It is in deprecated code that will be removed. There is no point in optimizing dead or dying code.

Reply:
Is `itertools.product` deprecated or only its pickling?

I did have other objections to this change. The code above and below use `Py_BuildValue()`, this change would make this line too different from other cases, which will harm readability. And this code is not performance critical anyway.